### ISSUE TITLE
Support redis password

### ISSUE THREAD

**Issue by ar4s** [django/asgi_redis#40]
Pasword from ``CHANNEL_LAYERS`` is not respected in ``txredisapi.ConnectionPool``.

For example:
```
CHANNEL_LAYERS = {
    'default': {
        'BACKEND': 'asgi_redis.RedisChannelLayer',
        'CONFIG': {
            'hosts': [
                'redis://:pass@host:port/2'
            ],
         },
        'ROUTING': 'foo.routing.channel_routing', 
    },
}
```
With this config ``daphne`` returns exception like: 
```Error trying to receive messages: Redis error: could not set dbid=2: NOAUTH Authentication required.```

**Comment by proofit404** [comment]
Actually channel layer just call `Redis.from_url` under the hood. Isn't it necessary to pass username?

**Comment by ar4s** [comment]
@proofit404 ok, but I have problem when I use ``channels`` with ``twisted`` and ``txredisapi``?

**Comment by proofit404** [comment]
Oh, sorry, I didn't notice it.

**Comment by andrewgodwin** [comment]
Fixed in #41.
